### ISSUE TITLE
Getter does multiple passes over data?

### ISSUE THREAD

**Issue by kcajf** [epezent/implot#420]
When using e.g. `PlotScatterG`, it seems like the getter is called twice for each data point. Is this expected / documented?

**Comment by epezent** [comment]
Yes, that's expected. There are potentially three render passes that will iterate though then getter -- line, marker face, marker edge. 

What sort of issue is this causing you? (I assume you're updating state when your getter is called?)
